demo/app.py

```diff
@@ -84,6 +87,13 @@ def gen_button_clicked():
 
     col1, col2 = st.columns(2)
 
+    with NamedTemporaryFile(delete=False, suffix=extension) as tmp_file:
+        tmp_file.write(uploaded_file.getvalue())
+        tmp_file_path = tmp_file.name
+
+    raw_text = DATA_LOADERS[extension](tmp_file_path)
+    os.unlink(tmp_file_path)
```

Comment:
If you move the code above into `load_file`, then you can remove line 94 and also move line 95 inside the `load_file` function at the end.

Reply:
Gotcha!

Reply:
Moved to the `load_file` function and added better exception handling.